mlir/tools/mlir-rocm-runner/mlir-rocm-runner.cpp

```diff
@@ -64,7 +69,9 @@ static LogicalResult runMLIRPasses(ModuleOp m) {
   pm.addPass(createGpuKernelOutliningPass());
   auto &kernelPm = pm.nest<gpu::GPUModuleOp>();
   kernelPm.addPass(createStripDebugInfoPass());
-  kernelPm.addPass(createLowerGpuOpsToROCDLOpsPass(/*indexBitWidth=*/32));
+  if (!rocdlInput.getValue()) {
+    kernelPm.addPass(createLowerGpuOpsToROCDLOpsPass(/*indexBitWidth=*/32));
+  }
```

Comment:
There is already one such option below

https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/ba41417f597092a1d379ffe3deb929d4f24e4402/mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp#L2515-L2519

Reply:
That's in the other context - the output of that can't be passed to mlir-rocm-runner

Reply:
Okay just to make sure I understand: This is to make sure you can directly execute a test case written in gpu dialect?

If that's the case why not use mlir-opt or mlir-miopen-driver to pipe through only the gpu->rocdl pass and then pipe it to mlir-rocm-runner to execute?

I'm not convinced that we should further branch the source dialect of `mlir-rocm-runner` because we already have tool to be able to handle the lowering part. The runner's responsibility is to be able to execute using a lowest possible dialect.

Reply:
The goal of this option is something that came up during debugging the LLVM tip changes. I want to be able to execute a GPU kernel that's been run through mlir-opt, even if it's not _our_ mlir-opt.

Reply:
But doesn't gpu->rocdl ops pass available in upstream?

If somehow the upstream version of gpu->rocdl pass isn't working I'm fine with this as a temporary change but I'd ask for a comment on this new command line option that once this pass is available upstream we should get rid of it.

Reply:
Oh, yeah, gpu->rocdl is available elsewhere. The problem is that, if I have a kernel in ROCDL instead of in GPU, I don't have a way to execute it with the tooling we have. Adding an option to skip the GPU->ROCDL translation seems like the right way to do that to me.

Reply:
Now without the context the assertion is true for every intermediate dialect. Taking an extreme example, if you have a kernel in top level MIOpen dialect, you can't directly use `mlir-rocm-runner` to execute it as well, but we don't want to make `mlir-rocm-runner` to do all the job that should belong to `mlir-miopen-driver`.

Reply:
Except ROCDL is below GPU. Your logic could show that maybe mlir-rocm-runner shouldn't be calling the kernel->ROCDL passes at all and leaving those to mlir-miopen-driver

Reply:
We clarified offline and think rocdl dialect is a more reasonable ir to pass to the runner. However this may result in a long discussion about what the responsibility of driver/runner actually are. So for now we should enable both capabilities for the development needs to evolve as we go.